mjpc/tasks/humanoid/tracking/task.cc

```diff
@@ -191,10 +243,35 @@ void humanoid::Tracking::Residual(const double *parameters,
 // ----------------------------------------------------------------------------
 void humanoid::Tracking::Transition(const mjModel *model, mjData *d,
                                     Task *task) {
-  // Hardcoded constant matching keyframes from CMU mocap dataset.
+
+  // hardcoded constant matching keyframes from CMU mocap dataset.
   float fps = 30.0;
-  double current_index = d->time * fps;
-  int last_key_index = (model->nkey) - 1;
+
+   
+  // get motion trajectory length
+  int length = TrajectoryLength(task->stage);
+
+  // get motion start index 
+  int start = MotionStartIndex(task->stage);
+
+  // check for motion switch 
+  if ((int)d->userdata[0] != task->stage || d->time == 0.0) {
+    d->userdata[0] = task->stage; // set motion id
+    d->userdata[1] = d->time;  // reference time
+
+    // set home state
+    double* qpos_home = model->key_qpos + model->nq * start;
+    double* qvel_home = model->key_qvel + model->nv * start;
+
+    mju_copy(d->qpos, qpos_home, model->nq);
+    mju_copy(d->qvel, qvel_home, model->nv);
+
+    printf("Motion Switch\n");
```

Comment:
Maybe print the motion index/name here too.

Reply:
Done.